bubble_sort.rb

```diff
@@ -0,0 +1,45 @@
+def bubble_sort(new_array)
+  flag = 0
+  while flag.zero?
+    flag = 1
+    new_array.each_with_index do |numb, position|
+      next if position == new_array.length - 1
+
+      test = yield(numb, new_array[position + 1])
+      next unless test.positive?
+
+      new_array[position] = new_array[position + 1]
+      new_array[position + 1] = numb
+      flag = 0
+    end
+  end
+end
```

Comment:
- bubble_sort does not interact with any block. So yield is redundant. Instead, it should compare two items.
 - Sort returns the sorted array. So your bubble_sort should do the same task. But it is not returning anything. That's why while i tried to execute the following code it did not work.
```ruby
print bubble_sort([1, 5, 3, 2, 1])
```
So, after line 15 you should place new_array.

Reply:
We implemented the next changes

Added new_array to both methods, now it prints both results.
We fixed the redundant yield on bubble_sort method.

Thank you for your review.